vllm/v1/core/scheduler.py

```diff
@@ -310,6 +343,10 @@ def schedule(self) -> "SchedulerOutput":
                         self.encoder_cache_manager.allocate(request, i)
                     encoder_budget = new_encoder_budget
 
+        # Put back any skipped requests at the head of the waiting queue
+        if still_waiting:
+            self.waiting = still_waiting + self.waiting
```

Comment:
QQ: Isn't it faster to do `self.waiting.append_left`?

Reply:
<img width="451" alt="image" src="https://github.com/user-attachments/assets/29ecae2b-43f7-4abf-b384-1dc664e36ef1" />

+ is a bit faster than `extendleft`

Reply:
the test case

```python
def test_plus():
  waiting = deque(range(1000))
  waiting_for_fsm = deque(range(100))
  waiting = waiting_for_fsm + waiting
  return waiting
  
def test_extendleft():
  waiting = deque(range(1000))
  waiting_for_fsm = deque(range(100))
	waiting.extendleft(waiting_for_fsm)
	return waiting
```

<img width="368" alt="image" src="https://github.com/user-attachments/assets/3377cf8a-97fa-4a41-b0fc-a35e25b8a304" />

Seems like extendleft is a bit faster comparing to plus operation

Reply:
@aarnphm `extendleft()` is not what you want here. It reverses the order of elements.

https://docs.python.org/3/library/collections.html#collections.deque.extendleft

Reply:
btw it's really surprising to me that it's faster since `extendleft()` sounds like it's having to move elements 1 at a time, where the addition operator seems like it's just updating a few pointers:

roughly ...

list1.tail.next = list2.head
list2.head.prev = list1.tail
list.1.tail = list.2.tail
return list1